website/content/tutorials/3-constraints-expect/_index.md

```diff
@@ -0,0 +1,1511 @@
+    address: 1000 K Street NW Washington, DC 20001
+    website: https://example.com/awesomecomp/
+  motherboard:
+    vendor:
+      id: vendor2
+      name: ISA Corp
+      address: 2000 K Street NW Washington, DC 20002
+      website: https://example.com/isacorp/
+    product-name: Magestic Model M-Ultra Motherboard
+    type: atx
+    cpu:
+      vendor:
+        id: vendor2
+        name: ISA Corp
+        address: 2000 K Street NW Washington, DC 20002
+        website: https://example.com/isacorp/
+      architecture: arm64
+      product-name: Superchip Model 1 4-core Processor
+      speed: 4.7 gigahertz
+    ata-socket:
+      vendor:
+        id: vendor2
+        name: ISA Corp
+        address: 2000 K Street NW Washington, DC 20002
+        website: https://example.com/isacorp/
+      product-name: AwesomeSAS Model 1 Storage Socket
+      type: sas
+    memory-modules:
+    - vendor:
+        id: vendor3
+        name: Erroneous Massive Memory Corp
+        address: 3000 K Street NW Washington, DC 20003
+        website: https://example.com/massive-memory-corp/
+      product-name: Model 3 DDR4-3200 8GB (Module 1)
+      byte-size: 8589934591
+    - byte-size: 8589934592
+      product-name: Model 3 DDR4-3200 8GB (Module 2)
+      vendor:
+        address: 3000 K Street NW Washington, DC 20003
+        id: vendor3
+        name: Massive Memory Corp
+        website: https://example.com/massive-memory-corp/        
+{{< /highlight >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+We can use the samples above to simulate a scenario where a developer or their tool define an invalid memory module configuration. The table below identifies constraints and their evaluation against the new sample. We can observe the processor should report on specific constraint errors for some fields, but not others.
+
+| Constraint ID             | Evaluated Target                                                  | Evaluated Test            | Result |
+|---------------------------|-------------------------------------------------------------------|---------------------------|--------|
+| memory-divisible-two      | `<!-- 1st memory module -->`<br/>`<byte-size>8589934591</byte-size>` |  `(8589934591 mod 2) = 0`    | false  |
+|                           | `<!-- 2nd memory module -->`<br/>`<byte-size>8589934592</byte-size>` |  `(8589934592 mod 2) = 0`    | true   |
+| memory-divisible-megabyte | `<!-- 1st memory module -->`<br/>`<byte-size>8589934591</byte-size>` |  `(8589934591 mod 1024) = 0` | false  |
+|                           | `<!-- 2nd memory module -->`<br/>`<byte-size>8589934592</byte-size>` |  `(8589934592 mod 1024) = 0` | true   |
+| memory-same-byte-size     | Sequence of <br/> (`<memory>...<product-name>... (Module 1)</product-name><byte-size>8589934591</byte-size></memory>`, <br/>`<memory>...<product-name>... (Module 2)</product-name><byte-size>8589934592</byte-size></memory>`) | `(sum((8589934592, 8589934592)) mod 8589934592) = 0)` |  true |
```

Comment:
Is there a reason this table is duplicated from above?

Reply:
If you look some the values are changing from the math and the results column accordingly switches from true to false. I just wanted to start explaining how a developer could conceptualize "what is the Metapath doing and why?" match by match and eval by eval.